src/components/admin/form-components/DirectTextField.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import TextField from 'material-ui/TextField';
+
+export class DirectTextField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onChange = event => props.onUpdate(event.target.value);
+  }
+
+  render() {
+    const passedProps = _.omit(this.props, 'onUpdate');
+    return <TextField onChange={this.onChange} {...passedProps} />;
+  }
+}
+
+DirectTextField.propTypes = {
+  onUpdate: PropTypes.func.isRequired,
```

Comment:
Do you think we can do `...TextField.propTypes` after this? I think it'd work and it'd be super neat

Reply:
@emilgoldsmith that is not strictly needed, I believe. Isn't the `TextField.propTypes` already checked for by the virtue of calling `<TextField ... />` with the passed props? Wouldn't it be better code separation to leave the propType enforcement up to the composition, rather than hard-binding them in the DirectTextField?

Reply:
@frederikbrinck I think you're right in terms of the functionality of the `propTypes` but propTypes are not only for the runtime prop validation, it's also for documentation for future developers, and if someone wants to use this component in the future I think it's nice that they can just look at the propTypes and see that they also need to pass through the props that will go to `TextField`, otherwise they'll have to look at the source to understand that (it's a small file so maybe not that big of a difference but still). Otherwise I feel like the propTypes are indicating that it's going to handle creating all the props needed for whatever children it has and you as a developer don't have to pass it anything but an `onUpdate` function

Reply:
I agree that it is for documentation purposes, but I would rather add a comment saying that it takes all the propTypes of the textField. Especially because as we build more and more complex components, we will end up having a huge list of propTypes for a component that composes a lot of subcomponents. And this list will be mostly replicated. Anyway, I don't expect our components to grow that fast in the future so doing what you say could work short term too.

Reply:
I see your point, I think I still politely disagree though ;).

> but I would rather add a comment saying that it takes all the propTypes of the textField

So you want to split up where the developer looks for what props this component takes into 2 places, now the developer both needs to look at a comment and at the proptypes?

> Especially because as we build more and more complex components, we will end up having a huge list of propTypes for a component that composes a lot of subcomponents

I don't think so, this pattern is only for wrapper components that pass all the rest of it's props down to a single child component. Normally a component takes very specific props, and does computations on them / does it's own computations based on state etc. and then pass computed props / hardcoded props on to children. The only reason we're doing this now is because all this component does is standardize the onChange API and then passes all the rest of it's props to TextField.

> what you say could work short term too.

Because of what I said above I think this is definitely also the long term solution, and the way to go. Feel free to prove me wrong though if I went wrong somewhere!

Reply:
Also I realized that if we don't do this it also makes error messages worse. As the prop types error will be that TextField was given incorrect props, when the problem was actually that you passed wrong props to DirectTextField. Maybe a small thing that you'll figure out quickly as it'll probably have the stack trace, but why make life harder for the developer for no reason when this solution seems super elegant?